### PR TITLE
Remove non alphanumeric characters from anchors

### LINKED ISSUE #203: gen-resourcesdocs, Markdown anchor multi-word text
This text is displaying on the page:
`Environment variables {#Environment variables}`
https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/ephemeral-container/#environment-variables-environment-variables

https://kubernetes.io/docs/reference/kubernetes-api/workloads-resources/ephemeral-container/#not-allowed-not-allowed

Fix the anchor text. Add hyphens between words.

### PR DESCRIPTION
**Pull request by feloy**
Fixes #203 by replacing non alphanumeric characters by hyphens in anchors

### CODE REVIEW

File: gen-resourcesdocs/templates/chapter-single-definition.tmpl [hunk 1]
```diff
@@ -29,7 +29,7 @@ weight: {{.Metadata.Weight}}
 {{- end}}{{/* range .Fields */}}
 
 {{range .FieldCategories}}
-### {{.Name}} {#{{.Name}}}{{/* explicitly set fragment to keep capitalization */}}
+### {{.Name}} {#{{"-" | regexReplaceAll "[^a-zA-Z0-9]+" .Name }}}{{/* explicitly set fragment to keep capitalization */}}
```
Comment (kbhawkey) [inline gen-resourcesdocs/templates/chapter-single-definition.tmpl:32]: @feloy, I am not familiar with this method(from `masterminds.github.io/sprig/strings.html`)?
Are all characters, beside alphanumerical characters, replaced with a hyphen?
In other words, "Not     allowed" --> "Not-----allowed"?

Comment (feloy) [reply gen-resourcesdocs/templates/chapter-single-definition.tmpl:32]: More precisely, all series of non alphanumeric characters is replaced by an hyphen. For example
```
Local / Temporary Directory {#Local-Temporary-Directory}
```

**Comment by kbhawkey** [comment]
Thanks @feloy .
/lgtm
/approve

**Pull request merged**